apps/src/lib/node/ledger/shell/init_chain.rs

```diff
@@ -92,9 +92,10 @@ where
             storage,
         } in genesis.established_accounts
         {
-            let vp_code = vp_code_cache
-                .get_or_insert_with(vp_code_path.clone(), || {
+            let vp_code =
+                vp_code_cache.get_or_insert_with(vp_code_path.clone(), || {
                     wasm_loader::read_wasm(&self.wasm_dir, &vp_code_path)
+                        .unwrap()
```

Comment:
I guess in here the error should really propagate up to the `init_chain`'s Result

Reply:
Yeah it would be good if this could propagate up, I had a go at this though it was more involved than I thought at a glance
- changed the code to reading the wasm outside of the closure so we can return the `Result` if there is an error
- had to make a new `shell::Error::ReadingWasm` variant to wrap the `eyre::Error`
There are a couple more occurrences in `init_chain` I could update to use the same but could be worth doing all these changes in a separate PR (I think a `CodeCache` component wrapping `HashMap<String, Vec<u8>>` and handling wasm loading could be useful)

Reply:
Thanks! Yeah, let's refactor the rest of them separately. Btw, I didn't know about the `#[from]` macro, very convenient!